datafusion/physical-plan/src/filter.rs

```diff
@@ -200,15 +200,10 @@ impl ExecutionPlan for FilterExec {
             // assume filter selects 20% of rows if we cannot do anything smarter
             // tracking issue for making this configurable:
             // https://github.com/apache/arrow-datafusion/issues/8133
-            let selectivity = 0.2_f32;
+            let selectivity = 0.2_f64;
```

Comment:
Interestingly this code from @andygrove  in https://github.com/apache/arrow-datafusion/pull/8126 is different to the way the selectivity is implemented below -- it uses `f32` and doesn't apply `ceil()` 

This PR makes sure these two paths are consistent which I think is an improvement